link-parent/link-ilp-over-http/src/main/java/org/interledger/link/http/IlpOverHttpLinkFactory.java

```diff
@@ -0,0 +1,106 @@
+    this.okHttpClient = Objects.requireNonNull(okHttpClient);
+    this.decryptor = Objects.requireNonNull(decryptor);
+    this.objectMapper = Objects.requireNonNull(objectMapper);
+    this.ilpCodecContext = Objects.requireNonNull(ilpCodecContext);
+  }
+
+  /**
+   * Construct a new instance of {@link Link} using the supplied inputs.
+   *
+   * @return A newly constructed instance of {@link Link}.
+   */
+  public Link<?> constructLink(
+      final Supplier<Optional<InterledgerAddress>> operatorAddressSupplier, final LinkSettings linkSettings
+  ) {
+    Objects.requireNonNull(linkSettings);
+
+    if (!this.supports(linkSettings.getLinkType())) {
+      throw new RuntimeException(
+          String.format("LinkType `%s` not supported by this factory!", linkSettings.getLinkType())
+      );
+    }
+
+    // Translate from Link.customSettings, being sure to apply custom settings from the incoming link.
+    final ImmutableIlpOverHttpLinkSettings.Builder builder = IlpOverHttpLinkSettings.builder().from(linkSettings);
+    final IlpOverHttpLinkSettings ilpOverHttpLinkSettings =
+        IlpOverHttpLinkSettings.applyCustomSettings(builder, linkSettings.getCustomSettings()).build();
+
+    final BearerTokenSupplier bearerTokenSupplier;
+    if (ilpOverHttpLinkSettings.outgoingHttpLinkSettings().authType().equals(IlpOverHttpLinkSettings.AuthType.SIMPLE)) {
+      // Decrypt whatever is inside of the encryptedTokenSharedSecret. For the SIMPLE profile, this will decrypt to the
+      // actual bearer token.
+      bearerTokenSupplier = new SimpleBearerTokenSupplier(new String(
+          decryptor.decrypt(ilpOverHttpLinkSettings.outgoingHttpLinkSettings().encryptedTokenSharedSecret().getBytes())
+      ));
+    } else {
+      // TODO: For now, we assume the bytes are a String that conform to the Crypt CLI. However, this should be made
+      // type-safe and more generic if possible. E.g., CryptoCLI formate vs Protobuf. Or, standardize on a single
+      // type-safe format?
+
+      // NOTE: This supplier will always create a copy of the decrypted bytes so that the consumer of each call can
+      // safely wipe the bytes from memory without affecting other callers.
+      final SharedSecretBytesSupplier sharedSecretSupplier = () -> decryptor
+          .decrypt(ilpOverHttpLinkSettings.outgoingHttpLinkSettings().encryptedTokenSharedSecret().getBytes());
+
+      bearerTokenSupplier = new JwtHs256BearerTokenSupplier(
+          sharedSecretSupplier, ilpOverHttpLinkSettings.outgoingHttpLinkSettings()
+      );
+    }
+
+    final IlpOverHttpLink ilpOverHttpLink = new IlpOverHttpLink(
+        operatorAddressSupplier,
+        ModifiableIlpOverHttpLinkSettings.create().from(ilpOverHttpLinkSettings), // Modifiable for testing
+        okHttpClient,
+        objectMapper,
+        ilpCodecContext,
+        bearerTokenSupplier
+    );
+
+    return ilpOverHttpLink;
```

Comment:
Maybe we could directly `return new IlpOverHttpLink()`? Is there a specific reason for this approach? Do we envision this to be changed and used for some kind of validation or checks before returning?

Reply:
No specific reason - it's likely that at some point during development I might have been doing something before line 98, but _after_ linke 89. As-is though, we can certainly change this.